topdown/http.go

```diff
@@ -139,12 +139,22 @@ func addHeaders(req *http.Request, headers map[string]interface{}) (bool, error)
 	for k, v := range headers {
 		// Type assertion
 		header, ok := v.(string)
-		if ok {
+		if !ok {
+			return false, fmt.Errorf("invalid type %T for headers value", v)
+		}
+
+		// If the Host header is given, bump that up to
+		// the request. Otherwise, just collect it in the
+		// headers.
+		k := http.CanonicalHeaderKey(k)
+		switch k {
+		case "Host":
+			req.Host = header
+		default:
 			req.Header.Add(k, header)
```

Comment:
🗒 Before, this `k` wasn't passed through `http.CanonicalHeaderKey`'s canonicalization; however, I've tried it locally, it makes no difference. The outgoing request's header keys end up being canonicalized. ✔️

Reply:
It makes sense that the HTTP client would canonicalize everything before it sends, though I don't know it that's guaranteed. I ended up with this code because I had to canonicalize to check for the Host and then I already had the canonical value in hand :)